app/frontend/src/submissions/views/SubmissionsHome.vue

```diff
@@ -579,9 +582,118 @@ export default {
+      }
+      if (owner_postal_code && !isValidPostalCodeOrZipCode(owner_postal_code)) {
+        errors.owner_postal_code = ['Invalid Postal Code or ZIP Code.'];
+      }
+    },
+    validateWellIdentificationPlateFields(errors) {
+      const { 
+        well_class,
+        identification_plate_number,
+        well_identification_plate_attached,
+      } = this.form
+
+      const validateWellClasses = [WELL_CLASS.WATER_SUPPLY, WELL_CLASS.INJECTION, WELL_CLASS.RECHARGE]
+      const isWellIdentificationPlateToBeVerified = validateWellClasses.includes(well_class)
+      
+      if (isWellIdentificationPlateToBeVerified == false) { return; }
+
+      if (!identification_plate_number) {
+        errors.identification_plate_number = ['Identification Plate Number Required.'];
+      }
+      if (!well_identification_plate_attached) {
+        errors.well_identification_plate_attached = ['Well Identification Plate Attached Required.'];
+      }
+    },
+    validateWellFields(errors) {
+      const { 
+        work_start_date, 
+        work_end_date, 
+        drilling_methods, 
+        total_depth_drilled, 
+        finished_well_depth
+      } = this.form;
+
+      if (!work_start_date) {
+        errors.work_start_date = ['Start Date of Work Required.'];
+      }
+      if (!work_end_date) {
+        errors.work_end_date = ['End Date of Work Required.'];
+      }
+      if (drilling_methods.length === 0) {
+        errors.drilling_methods = ['Drilling Methods Required.'];
+      }
+      if (!total_depth_drilled) { 
+        errors.total_depth_drilled = ['Total Depth Drilled Required.']; 
+      }
+      if (!finished_well_depth) { 
+        errors.finished_well_depth = ['Finished Well Depth Required.']; 
+      }
+    },
+    newlyConstructedWellValidation(errors) {
+      const { 
+        work_start_date,
+        work_end_date,
+      } = this.form
+      
+      const mandatoryLicensingDate = new Date(Date.UTC(
+        NEW_WELL_CONSTRUCTION_VALIDATION_DATE.YEAR,
+        NEW_WELL_CONSTRUCTION_VALIDATION_DATE.MONTH,
+        NEW_WELL_CONSTRUCTION_VALIDATION_DATE.DAY
+        )).getTime();
+
+      const workStartDate = getUTCDate(work_start_date);
+      const workEndDate = getUTCDate(work_end_date);
+
+      if (this.activityType !== TYPE_OF_WORK.CON) { return; }
+
+      if ((!isNaN(workStartDate) && !isNaN(workEndDate)) && workStartDate > workEndDate) {
+        errors.work_start_date = ['Invalid Start Date comes after End Date.'];
+        errors.work_end_date = ['Invalid End Date comes before Start Date.'];
+      }
+
+      if (!isNaN(workEndDate) && workEndDate >= mandatoryLicensingDate) {
+        this.validateWellIdentificationPlateFields(errors);
+        this.validateWellFields(errors);
+        return
+      }
+
+      if (!isNaN(workStartDate) && workStartDate >= mandatoryLicensingDate) {
+        this.validateWellIdentificationPlateFields(errors);
+        this.validateWellFields(errors);
+        return
```

Comment:
```suggestion
```
Sonarcloud actually flagged this one since its there's no code following it, making it redundant.
But if its scalability in mind that's another thing entirely  \o/

Reply:
You and Sonarcloud are right I've updated for clarity.